service/app/services/BillService.scala

```diff
@@ -76,6 +80,42 @@ class BillService @Inject()(repository: BillRepository,
   def updateBillCalculation(bill: Bill): Future[Int] =
     repository.modify(bill.id.get, bill)
 
+  def updateAddress(billId: Long,
+                    address: String,
+                    longitude: Double,
+                    latitude: Double): Future[Option[Bill]] =
+    retrieveOne(billId).flatMap {
+      case Some(bill) =>
+        repository.updateAddress(billId, address, longitude, latitude).flatMap {
+          case 1 =>
```

Comment:
Not fan of hard-coded values. Maybe it's better to check it with simple `if` condition.